nomenclature/codelist.py

```diff
@@ -221,7 +192,29 @@ def from_directory(cls, name: str, path: Path, file: str = None):
         instance of cls (CodeList if not inherited)
 
         """
-        return cls(name=name, mapping=cls._parse_dir(name, path, file))
+        code_list = []
+        # parse all files in path if file is None
+        file = file or "**/*"
```

Comment:
Same comment as above regarding the name of the `file` parameter. Also you could just add "**/*" as a default argument and remove this line.